<launch>

  <!-- Set use_sim_time true for datasets-->
  <param name="use_sim_time" value="true"/>

  <!-- Clear old S-PTAM parameters, just in case -->
  <rosparam command="delete" ns="sptam"/>

  <!-- Read S-PTAM parameters file -->
  <rosparam command="load" ns="sptam" file="$(find sptam)/configurationFiles/mit.yaml" />

  <!-- PR2 robot model (for visualization) -->
<!--  <include file="$(find pr2_description)/robots/upload_pr2.launch" />-->

  <group ns="wide_stereo">
    <!-- Call stereo_image_proc to undistort and rectify images -->
    <node pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc">
      <remap from="/stereo/left/image_raw"  to="/wide_stereo/left/image_raw"/>
      <remap from="/stereo/right/image_raw"  to="/wide_stereo/right/image_raw"/>
      <remap from="/stereo/left/camera_info"  to="/wide_stereo/left/camera_info"/>
      <remap from="/stereo/right/camera_info"  to="/wide_stereo/right/camera_info"/>
    </node>
  </group>

  <!-- path publisher. Transforms poses into path. -->
  <!--
  <node pkg="ros_utils" type="pose_to_path" name="sptam_path">
    <remap from="pose" to="sptam/robot/pose" />
    <remap from="path" to="sptam/robot/path" />
  </node>
  -->

   <!-- Nodelet Manager -->
  <node pkg="nodelet" type="nodelet" name="nodelet_manager"  args="manager" output="screen" clear_params="true"/>

  <!-- SPTAM Nodelet -->
  <node pkg="nodelet" type="nodelet" name="sptam" args="load sptam/SPTAMNodelet nodelet_manager" output="screen" clear_params="true" >

    <param name="approximate_sync" value="false" />
    <param name="use_odometry" value="true"/>
    <param name="odom_frame" value="odom_combined"/>
    <param name="camera_frame" value="wide_stereo_gazebo_l_stereo_camera_optical_frame"/>

    <!-- Remap topics  -->
    <remap from="/stereo/left/image_rect" to="/wide_stereo/left/image_rect_color"/>
    <remap from="/stereo/right/image_rect" to="/wide_stereo/right/image_rect_color"/>
    <remap from="/stereo/left/camera_info" to="/wide_stereo/left/camera_info"/>
    <remap from="/stereo/right/camera_info" to="/wide_stereo/right/camera_info"/>

    <!-- Remap Odometry message -->
    <remap from="robot/pose"  to="odom"/>

  </node>

  <!-- DENSE node -->
  <node pkg="nodelet" type="nodelet" name="dense" args="load dense/denseNodelet nodelet_manager" output="screen" clear_params="true">
    <param name="disp_calc_method" value="libelas"/>
	<param name="min_disparity" value="10"/>
    <param name="voxel_leaf_size" value="0.03"/>

	<param name="stereoscan_threshold" value="1"/>
    <!--param name="fusion_heuristic" value="simpleMean"/-->
    <param name="fusion_heuristic" value="weigthDistances"/>

	<param name="frustum_near_plane_dist" value="0.0001"/>

	<param name="add_corners" value="false"/>
	<param name="libelas_ipol_gap" value="5000"/>

	<param name="local_area_size" value="10"/>

    <remap from="/dense/keyframes" to="/sptam/keyframes"/>

  </node>

</launch>
